plugins/inputs/proxmox/proxmox.go

```diff
@@ -137,7 +139,12 @@ func gatherVmData(px *Proxmox, acc telegraf.Accumulator, rt ResourceType) {
 			return
 		}
 		tags := getTags(px, vmStat.Name, vmConfig, rt)
-		fields, err := getFields(vmStat)
+		currentVMStatus, err := getCurrentVMStatus(px, rt, vmStat.ID)
```

Comment:
Just a note that this adds an http request inside a loop. It's going to have some negative performance implications, especially on systems with a large number of VMs. If there's any way to get this data in aggregate with one call, that would be preferable.

Reply:
We are aware of this, but AFAIK there is no other possibility in the REST-API to query the status of all virtual machines containers in one step (please see: https://pve.proxmox.com/pve-docs/api-viewer/)